### PR TITLE
Move in face detection sample + test

### PR DESCRIPTION
**Pull request by jerjou**

### CODE REVIEW

File: vision/api/face_detection/faces.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from PIL import Image
+from PIL import ImageDraw
+
+from googleapiclient import discovery
+import httplib2
+from oauth2client.client import GoogleCredentials
+
+
+# [START get_vision_service]
+DISCOVERY_URL='https://{api}.googleapis.com/$discovery/rest?version={apiVersion}'
```
Comment (jonparrott) [inline vision/api/face_detection/faces.py]: Is this still needed?

File: vision/api/face_detection/faces.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+# limitations under the License.
+
+"""Draws squares around faces in the given image."""
+
+import argparse
+import base64
+
+from PIL import Image
+from PIL import ImageDraw
+
```
Comment (jonparrott) [inline vision/api/face_detection/faces.py]: Both of these sections are third-party imports, no need for a newline.

File: vision/api/face_detection/faces.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    Args:
+        face_file: A file-like object containing an image with faces.
+
+    Returns:
+        An array of dicts with information about the faces in the picture.
+    """
+    image_content = face_file.read()
+    batch_request = [{
+        'image': {
+            'content': base64.b64encode(image_content).decode('UTF-8')
```
Comment (jonparrott) [inline vision/api/face_detection/faces.py]: `utf-8`, please.

File: vision/api/face_detection/faces.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          have polygons drawn around them.
+    """
+    im = Image.open(image)
+    draw = ImageDraw.Draw(im)
+
+    for face in faces:
+        box = [(v.get('x', 0.0), v.get('y', 0.0)) for v in face['fdBoundingPoly']['vertices']]
+        draw.line(box + [box[0]], width=5, fill='#00ff00')
+
+    del draw
```
Comment (jonparrott) [inline vision/api/face_detection/faces.py]: Is this necessary? It'll be GCd after this function?

File: vision/api/face_detection/faces.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    im.save(output_filename)
+# [END highlight_faces]
+
+
+# [START main]
+def main(input_filename, output_filename, max_results):
+    with open(input_filename, 'rb') as image:
+        faces = detect_face(image, max_results)
+        print('Found %s face%s' % (len(faces), '' if len(faces) == 1 else 's'))
```
Comment (jonparrott) [inline vision/api/face_detection/faces.py]: use `.format` here and in the line below please.

File: vision/api/face_detection/faces_test.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import re
+import shutil
+import tempfile
+
+import pytest
+from PIL import Image
+from faces import main
+
+
+@pytest.yield_fixture
```
Comment (jonparrott) [inline vision/api/face_detection/faces_test.py]: pytest has this built-in: https://pytest.org/latest/builtin.html#builtin-fixtures-function-arguments

File: vision/api/face_detection/faces.py (hunk 1 above)
Comment (jerjou) [inline vision/api/face_detection/faces.py]: Unfortunately, yes - OP apis' endpoints are a subdomain of googleapis.com, whereas apiary ones aren't

**Comment by jerjou** [comment]
Updated - PTAL

**Comment by jonparrott** [comment]
LGTM, but you have to make travis happy first.

**Comment by jerjou** [comment]
I SHAKE MY FIST AT YOU TRAVIS (though to be fair it's my own fault...)

**Pull request merged**